servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -2664,6 +2664,10 @@ public final Publisher<T> skipWhile(Predicate<? super T> predicate) {
         return filter(FilterPublisher.skipWhileSupplier(predicate));
     }
 
+    public final <R> Single<R> splice(BiFunction<T, Publisher<T>, R> packer) {
```

Comment:
Please add a javadoc.

Wrt the name, it seems like what it does is processes the head of the publisher into something and then gives you the tail which I know as the `uncons` operator but I think that is more of an FP term. It's not a great name, but I think it's a tiny bit better since you're right: we don't actually splice anything here.

Reply:
duh, I missed this... Yeah, `uncons` is exactly what I was thinking but I wasn't sure if we wanted to go down that road...

Reply:
TIL about `uncons` :) 
I'm +1 for renaming `splice` to something. Reading about FP, `uncons` sometimes is also named `split`. But `split` can happen for any condition, not just the first elements like `uncons`. Some options to vote for:
- `uncons` - like in FP
- `splitFirst` - to indicate the condition
- `firstThenTail`/`firstAndTail` (or similar with `first*`) - similar to existing operators we already have: `firstOrElse` and `firstOrError`

Reply:
I think in terms of continuity with our codebase, something with `first` makes sense to me. I would place my vote with `firstAndTail`.

Reply:
`firstAndTail` is fine with me. 👍